### PR TITLE
fix: zendesk tickets must have a name

### PR DESCRIPTION
**Pull request by pauldambra**
see this twitter thread for context https://twitter.com/SHxKM/status/1759182599807713311

It turns out that users can get in a state in PostHog where they have an email but no name. This doesn't matter - except for submitting zendesk tickets. When submitting a zendesk ticket in a context zendesk considers anonymous the name must be present or the ticket will be rejected.

It was way too tricky to figure this out using existing error monitoring

1. alters how we send info to error monitoring so we can see this more quickly
2. always sets a value for name, we'd rather get a ticket that's hard to track down than frustrate a user

### CODE REVIEW

File: frontend/src/lib/components/Support/supportLogic.ts [hunk 1]
```diff
@@ -264,15 +265,22 @@ export const supportLogic = kea<supportLogicType>([
...
                     headers: { 'Content-Type': 'application/json' },
                 })
                 if (!response.ok) {
                     const error = new Error(`There was an error creating the support ticket with zendesk.`)
+                    const extra: Record<string, any> = { zendeskBody: zendeskRequestBody }
+                    Object.entries(payload).forEach(([key, value]) => {
+                        extra[`payload_${key}`] = value
+                    })
+                    extra['response_status'] = response.status
+                    extra['response_body'] = await response.text()
```
Comment (marandaneto) [inline frontend/src/lib/components/Support/supportLogic.ts:281]: There's a proper place to set request/response metadata, see [request](https://develop.sentry.dev/sdk/event-payloads/request/) and [response](https://develop.sentry.dev/sdk/event-payloads/contexts/#response-context) docs.
So the metadata is properly formatted and highlighted.

Comment (marandaneto) [reply frontend/src/lib/components/Support/supportLogic.ts:281]: Another option is to use this integration https://docs.sentry.io/platforms/javascript/configuration/integrations/httpclient/ which captures failed requests automatically with all its metadata (request and response).

**Review by Gilbert09 — APPROVED** [review]
Nice 🚢

File: frontend/src/lib/components/Support/supportLogic.ts (hunk 1 above)
Comment (marandaneto) [reply frontend/src/lib/components/Support/supportLogic.ts]: I didn't test the TS bits bit the data structure is correct https://github.com/PostHog/posthog/pull/20434/commits/a40100f428c65fed5ae0638f5a0f84c295903a3c
PS: I added [this](https://github.com/getsentry/rfcs/blob/main/text/0022-response-context.md) :D

**Review by marandaneto — APPROVED** [review]

**Pull request merged**